content/en/docs/reference/using-api/api-concepts.md

```diff
@@ -85,7 +85,60 @@ For example:
         }
         ...
 
-A given Kubernetes server will only preserve a historical list of changes for a limited time. Clusters using etcd3 preserve changes in the last 5 minutes by default.  When the requested watch operations fail because the historical version of that resource is not available, clients must handle the case by recognizing the status code `410 Gone`, clearing their local cache, performing a list operation, and starting the watch from the `resourceVersion` returned by that new list operation. Most client libraries offer some form of standard tool for this logic. (In Go this is called a `Reflector` and is located in the `k8s.io/client-go/cache` package.)
+A given Kubernetes server will only preserve a historical list of changes for a limited time. Clusters using etcd3 preserve changes in the last 5 minutes by default.  When the requested watch operations fail because the historical version of that resource is not available, clients must handle the case by recognizing the status code `410 Gone`, clearing their local cache, performing a list operation, and starting the watch from the `resourceVersion` returned by that new list operation. Most client libraries offer some form of standard tool for this logic. In Go this is called a `Reflector` and is located in the `k8s.io/client-go/cache` package. The following pseudo-code demonstrates how to implement a reflector:
```

Comment:
Reviewers: Is providing pseudo-code excessive here? Idea was to provide enough detail for client implementations in other languages.

Reply:
hmm... one the one hand, we need to communicate specifically how a client should correctly interact with this API. on the other, if it's this complex, will we have bugs in our pseudocode?

Reply:
It is complex. I'm not sure it work out to be much better than reading the go code. I'll drop this section and focus on getting the main section about resource version added.